alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -254,27 +261,49 @@ pattern Tx {body, wits, isValidating, auxiliaryData} <-
   where
     Tx b w v a = TxConstr $ memoBytes (encodeTxRaw $ TxRaw b w v a)
 
+-- We define these accessor functions manually, because if we define them using
+-- the record syntax in the Tx pattern, they inherit the constraints
+-- (Era era, ToCBOR (Core.AuxiliaryData era),ToCBOR (Core.TxBody era))
+-- constraint as a precondition. This is unnecessary, as one can see below.
+
+body' :: Tx era -> Core.TxBody era
+body' (TxConstr (Memo (TxRaw b _ _ _) _)) = b
```

Comment:
I would still be inclined to drop these, since they're not used (except as part of the `getField` definition). But it's not a hill to die on :-)

Reply:
these are useful in code like the pretty printer, where you just want the pieces without acquiring the extra constraint obligations. Future looking. Many future Eras will want to manipulate these fields (I am sure :-) )